pinot-common/src/main/proto/plan.proto

```diff
@@ -0,0 +1,74 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+syntax = "proto3";
+
+package org.apache.pinot.common.proto;
+
+message StageNode {
+  int32 stageId = 1;
+  string nodeName = 2;
+  repeated StageNode inputs = 3;
+  ObjectFields fields = 4;
```

Comment:
map<string, ObjectField> member_variables = 4

```
JoinRelType joinType;  -> mapEntry.of("joinType", LiteralField);
List<joinClause> joinClauses;  -> mapEntry.of("joinClauses", ListField);
```

ObjectFields fields = 4;

--> 
ObjectFields {
  map<string, ObjectField> memberVariableMap; // key is the member variable name, ObjectField is used to ser/deser the member variable
}

ObjectField {
one of {
  LiteralField
  ListField (List<ObjectField>)
  MapField this is a actual map member variable. 
  ObjectFields  {memberN1 --> ObjectField; memberN2 --> LiteralField; memberN3 --> ListField}
}
}

JOINNode
{
  ObjectFields (
    "joinType" --> LiteralField("BROADCAST"
    "joinClause" --> ListField<ObjectFields>(
      ObjectFields<JoinClause>(
        "leftJoinKey" --> ObjectFields<FieldSelectionKeySelector>(
          "columnIndex" --> LiteralField(1)
        )
        "rightJoinKey" --> ObjectFields<FieldSelectionKeySelector>(
          "columnIndex" --> LiteralField(1)
        ) 
  )
}


```
ConcreteStageNode implement StageNode {
  int a;
  List<Integer> b;
  InnerClass c;
  Map<string, Integer> d;
}

("a" --> ObjectField) --> LiteralObjectField --> 1
("b" --> ObjectField) --> ListObjectField --> List<Integer> --> [1, 2]
("c" --> ObjectField) --> MapObjectField --> ("memberVariableName" --> ObjectField) Map
("d" --> ObjectField) --> MapObjectField --> "random key" --> IntergerVariable)

Reply:
this has been implemented